test/Dockerfile.test-distroless

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### Dockerfile for building a distroless image for testing
+# Modified version of https://github.com/kubernetes/release/blob/master/images/build/distroless-iptables/distroless/Dockerfile
+
+ARG STAGE_DIR="/opt/stage"
+
+FROM debian:bullseye-slim as build
+ARG STAGE_DIR
+
+COPY test/build/stage-binaries-from-package.sh /
+COPY test/build/package-utils.sh /
+COPY test/build/stage-binary-and-deps.sh /
+
+RUN mkdir -p "${STAGE_DIR}" && \
+    /stage-binaries-from-package.sh "${STAGE_DIR}" conntrack \
+    ebtables    \
+    ipset       \
+    iptables    \
+    kmod		&& \
```

Comment:
Changed, it only installs iptables now.

I kept the installations scripts I copied because that was faster than re-rewriting the dpkg stuff myself, so the weird optimization is still there (I guess you were referring to filtering everything except libraries and copyrights?). Please, take a look to the new version and if this still feels too complicated I'll rewrite the staging script.